pkg/adsc/delta.go

```diff
@@ -295,6 +298,9 @@ func (c *Client) runOnce(ctx context.Context) error {
 }
 
 func (c *Client) dial(ctx context.Context) error {
+	if c.conn != nil {
+		_ = c.conn.Close()
```

Comment:
Any consequences to making dial destructive here? I'm specifically thinking about error handling. Before, if you called `dial()` and it errored, the existing connection would still be left in-tact. here, we close the connection before knowing if the new one succeeded

Reply:
From the current usage I don't think so since the  conn is always closed when we call this. But it is a bit sketchy I would say

Reply:
The code isn't fully wired up yet so I'm fine with this for now. Something to keep an eye on though